Add description of audio sample types

padenot opened the pull request:
Noteworthy things:
- Added `U8P`
- Added `S24` variant. Otherwise, there is no way to distinguish between 24bit audio sample and 32bit audio samples (this exists, in WAV). This follows what ffmpeg and others do.

Otherwise, probably what you'd have expected.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webcodecs/pull/256.html" title="Last updated on May 27, 2021, 9:04 AM UTC (3a8b026)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webcodecs/256/97e81bd...3a8b026.html" title="Last updated on May 27, 2021, 9:04 AM UTC (3a8b026)">Diff</a>

dalecurtis commented:
Technical details lgtm, but defer to @chcunningham for spec wording.

chcunningham commented on index.src.html:
```diff
@@ -2129,44 +2143,161 @@
+### Magnitude of the audio samples ### {#audio-samples-magnitude}
+
+The <dfn>minimum value</dfn> and <dfn>maximum value</dfn> of an audio sample,
+for a particular audio sample type, are the value below which
```
value -> values

chcunningham commented on index.src.html:
```diff
@@ -2129,44 +2143,161 @@
+The <dfn>minimum value</dfn> and <dfn>maximum value</dfn> of an audio sample,
+for a particular audio sample type, are the value below which
+(respectively above which) audio clipping might occur. They are otherwise regular
+types, that can hold values outside this interval, during intermeditate
```
nit: probably don't need commas here

chcunningham commented on index.src.html:
```diff
@@ -2129,44 +2143,161 @@
+</tbody>
+</table>
+
+Note: There is no data types that can hold 24 bits of information conveniently,
```
nit: types -> type

chcunningham commented on index.src.html:
```diff
@@ -2129,44 +2143,161 @@
+</table>
+
+Note: There is no data types that can hold 24 bits of information conveniently,
+but audio content using 24-bits samples is common, so 32-bits integers are
```
nit: *-bits -> *-bit

chcunningham commented on index.src.html:
```diff
@@ -2129,44 +2143,161 @@
+
+### Audio channel ordering ### {#audio-channel-ordering}
+
+When decoding, the ordering of the audio channels in an {{AudioData}} MUST be
```
I like how you used 'resulting' in the encoding rule. Suggest the following edit:

> When decoding, the ordering of the audio channels in the resulting AudioData MUST be the same as what is present in the given EncodedAudioChunk.

For the encoding rule, I would phrase closer to what you have for decoding... eg.

> When encoding, the ordering of the audio channels in the resulting EncodedAudioChunk MUST be the same as what is present in the given AudioData.

chcunningham reviewed (approved):
Really great! I only had nits.

padenot commented:
@chcunningham I've added another commit on top of the commit addressing the comments you had. Nothing controversial. The failure above will be fixed once I rebase on top of #228 (that itself needs to be rebased), because I'll reference the definition of a `frame`.

chcunningham commented:
Editors call: @aboba lgtm

padenot commented:
This is now rebased, and I've linked the concept of a frame in `copyTo` definitions and elsewhere, merging.
